### PR TITLE
Add shutdown-deferrer sidecar container to PODs

### PR DESCRIPTION
**Pull request by tuommaki**
Shutdown deferrer is a service that will monitor DrainerConfig object
for specific POD it's running in and prevents POD / container
termination before corresponding node is properly drained (by
node-operator).

Towards https://github.com/giantswarm/giantswarm/issues/2993

FYI: @giantswarm/sig-infrastructure

### CODE REVIEW

File: service/controller/v15/key/key.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ const (
 	K8SEndpointUpdaterDocker = "quay.io/giantswarm/k8s-endpoint-updater:df982fc73b71e60fc70a7444c068b52441ddb30e"
 	K8SKVMDockerImage        = "quay.io/giantswarm/k8s-kvm:db5f9c0ad08fd99a7e775e31fba99e5c7f87ab59"
 	K8SKVMHealthDocker       = "quay.io/giantswarm/k8s-kvm-health:ddf211dfed52086ade32ab8c45e44eb0273319ef"
+	ShutdownDeferrerDocker   = "quay.io/giantswarm/shutdown-deferrer:latest"
```
Comment (tuommaki) [inline service/controller/v15/key/key.go]: This is still very actively developed so hence `latest`.

File: service/controller/v15/resource/deployment/master_deployment.go [hunk 2]
```diff
@@ -321,6 +321,11 @@ func newMasterDeployments(customObject v1alpha1.KVMConfig) ([]*extensionsv1.Depl
 									},
 								},
 							},
+							{
+								Name:            "shutdown-deferrer",
+								Image:           key.ShutdownDeferrerDocker,
+								ImagePullPolicy: apiv1.PullAlways,
```
Comment (tuommaki) [inline service/controller/v15/resource/deployment/master_deployment.go]: There shall be environment variables added here later to pass POD information (via k8s Downward API mechanism) to sidecar.

**Review by r7vme — APPROVED** [review]
looks good

how do they will communicate k8s-kvm and shutdown-deferrer?

**Comment by xh3b4sd** [comment]
`shutdown-deferrer` provides an HTTP endpoint which `qemu-shutdown` is going to watch upon termination signals. The containers all share localhost within the pod so this should be fine somehow.

**Comment by calvix** [comment]
You will need changes in RBAC in order to watch CRDs

File: service/controller/v15/key/key.go (hunk 1 above)
Comment (tuommaki) [reply service/controller/v15/key/key.go]: Had to pin specific container version as we don't tag `latest`.

**Comment by tuommaki** [comment]
E2e on KVM still WIP. Tested cluster creation (from WIP) and upgrade from current active to WIP on `geckon` successfully. Going with this.

**Comment by tuommaki** [comment]
> You will need changes in RBAC in order to watch CRDs

I'll add these later when necessary (together with docker image upgrade).

**Pull request merged**